integration/benchmarks_test.go

```diff
@@ -178,3 +180,53 @@ func BenchmarkInsertMany(b *testing.B) {
 		}
 	}
 }
+
+func BenchmarkInsertManyIntoDifferentCollections(b *testing.B) {
+	const numCollections = 25
+
+	ctx, collection := setup.Setup(b)
+
+	provider := shareddata.BenchmarkSmallDocuments
+	iter := provider.NewIterator()
+
+	insertDocs := []any{}
+
+	for {
+		docs, err := iterator.ConsumeValues(iter)
+		if errors.Is(err, iterator.ErrIteratorDone) || docs == nil {
+			break
+		}
+
+		require.NoError(b, err)
+
+		for _, doc := range docs {
+			insertDocs = append(insertDocs, doc)
+		}
+	}
+
+	for i := 0; i < b.N; i++ {
+
+		var wg sync.WaitGroup
+		wg.Add(numCollections)
+		for ii := 0; ii < numCollections; ii++ {
+			go func(i int) {
+				r := rune('a' + i)
+				name := string(r)
+
+				err := collection.Database().CreateCollection(ctx, name)
+				require.NoError(b, err)
+
+				b.StartTimer()
+				collection.Database().Collection(name).InsertMany(ctx, insertDocs)
+				b.StopTimer()
+
+				err = collection.Database().Collection(name).Drop(ctx)
+				require.NoError(b, err)
+
+				wg.Done()
+			}(ii)
+
+		}
+		wg.Wait()
+	}
```

Comment:
What do you think about something like the following?

```go
b.StopTimer()
b.ResetTimer() // we don't care about time spent elsewhere
for i := 0; i < b.N; i++ {
	collections := make([]string, numCollections)
	
	for i := 0; i < numCollections; i++ {
		name := string(r)
		r := rune('a' + i)
		
		err := collection.Database().CreateCollection(ctx, name)
		require.NoError(b, err)
		collections = append(collections, collection.Database().Collection(name))
	}
	
	
	var wg sync.WaitGroup
	wg.Add(numCollections)

	b.StartTimer()
	for i := 0; i < numCollections; i++ {
		go func(i int) {
			collections[i].InsertMany(ctx, insertDocs)
			wg.Done()
		}(i)
	}
	b.StopTimer()
	wg.Wait()

	for i := 0; i < numCollections; i++ {
		err = colections[i].Drop(ctx)
		require.NoError(b, err)
	}

}
```

We start by stopping the timer (which starts automatically when *testing.B executes).
Then we reset it.
Then, we run our benchmark b.N times.

Instead of doing everything in a single step, we break it down in three steps:

1. Create all collections necessary for the benchmark.
2. Insert data into all collections.
3. Drop all collections.

We do this as we want to avoid measuring time spent creating collections and dropping them.
Even though they aren't measured directly in the original code thanks to `b.StopTimer()` and `b.StartTimer()` on the original code, they are still running in other operations in other goroutines and might negatively impact the performance.

By breaking down in three different parts it seems to me we'll measure a more accurate time.

Reply:
Without passing a value to `-benchtime` it crashes my machine.

Reply:
Here's a run with a value of 10 iterations. I could reduce the number of collections as well. I made it an array too, not a slice.

```
byron@byron-ThinkCentre-M910q:~/workdir/FerretDB/integration$ go test -race -benchmem -run=^$ -bench ^BenchmarkInsertManyIntoDifferentCollections$ -target-backend='ferretdb-postgresql' -postgresql-url='postgres://username@127.0.0.1:5432/ferretdb' -benchtime=10x
2024-01-30T19:11:11.336+0100	INFO	setup/startup.go:90	Target system: ferretdb-postgresql (built-in).
2024-01-30T19:11:11.336+0100	INFO	setup/startup.go:103	Compat system: none, compatibility tests will be skipped.
2024-01-30T19:11:11.345+0100	INFO	debug	debug/debug.go:103	Starting debug server on http://127.0.0.1:37501 ...
2024-01-30T19:11:11.345+0100	INFO	debug	debug/debug.go:109	http://127.0.0.1:37501/debug/graphs - Visualize metrics
2024-01-30T19:11:11.345+0100	INFO	debug	debug/debug.go:109	http://127.0.0.1:37501/debug/metrics - Metrics in Prometheus format
2024-01-30T19:11:11.345+0100	INFO	debug	debug/debug.go:109	http://127.0.0.1:37501/debug/pprof - Runtime profiling data for pprof
2024-01-30T19:11:11.346+0100	INFO	debug	debug/debug.go:109	http://127.0.0.1:37501/debug/vars - Expvar package metrics

goos: linux
goarch: amd64
pkg: github.com/FerretDB/FerretDB/integration
cpu: Intel(R) Core(TM) i5-6500T CPU @ 2.50GHz
BenchmarkInsertManyIntoDifferentCollections-4   	
number of active goroutines: 101 running for: 10.00 seconds
number of active goroutines: 136 running for: 10.00 seconds
number of active goroutines: 147 running for: 20.02 seconds
number of active goroutines: 100 running for: 20.01 seconds
^Cpanic: Stopping everything

goroutine 1160 [running]:
github.com/FerretDB/FerretDB/internal/util/testutil.Ctx.func2()
	/home/byron/workdir/FerretDB/internal/util/testutil/testutil.go:56 +0xf0
created by github.com/FerretDB/FerretDB/internal/util/testutil.Ctx in goroutine 1112
	/home/byron/workdir/FerretDB/internal/util/testutil/testutil.go:43 +0x2a7
exit status 2
FAIL	github.com/FerretDB/FerretDB/integration	27.850s
```

Reply:
Moved array creation [here](https://github.com/FerretDB/FerretDB/pull/3982/files#diff-ad672244bfcf1fbe3f5d2e2959e6c9b4bc0b0ff2f8f6b5b512c7db889881bca8R218) and it seems to perform better now but still consumes all memory:

```
byron@byron-ThinkCentre-M910q:~/workdir/FerretDB/integration$ free -m
               total        used        free      shared  buff/cache   available
Mem:           15751        7998         146         899        7606        6526
Swap:              0           0           0
```